back/app/services/invites_service.rb

```diff
@@ -262,13 +262,15 @@ def build_invites(hash_array, default_params={}, inviter=nil)
       invites = hash_array.map do |invite_params|
         build_invite(invite_params, default_params, inviter)
       end
-      # Since invites will later be created in a single transaction, the
-      # normal mechanism for generating slugs could result in non-unique
-      # slugs. Therefore we generate the slugs manually
+      # Since invites will later be created in a single transaction, the normal mechanism for
+      # generating slugs could result in non-unique slugs. Therefore, we generate the slugs manually,
+      # unless Abbreviated User Names feature is enabled.
       invitees = invites.map(&:invitee)
-      invitees.zip(SlugService.new.generate_slugs(invitees){|u| u.full_name}) do |(invitee, slug)|
-        if invitee.full_name.present?
-          invitee.slug = slug
+      if AppConfiguration.instance.feature_activated?('abbreviated_user_names')
+        invitees.each { |invitee| invitee.slug = SecureRandom.uuid if invitee.full_name.present? }
```

Comment:
Why do you set slug only if full_name is present?

Reply:
I hadn't actually thought that though. I mainly just mirrored the existing case. It might be a check that's only needed for the anonymised slug, since then the slug = `first_name+last_name`.

Or the check might have something to do with checking the names are there for use in the invitation email.

But you're probably right. I should check why, and if it matters.

Reply:
Yes, you're right. In `user.rb` we only care if `full_name` is present in non-anonymised case. I will edit accordingly.